src/function/scalar/map/map_keys.cpp

```diff
@@ -0,0 +1,78 @@
+#include "duckdb/planner/expression/bound_function_expression.hpp"
+#include "duckdb/common/string_util.hpp"
+#include "duckdb/parser/expression/bound_expression.hpp"
+#include "duckdb/function/scalar/nested_functions.hpp"
+#include "duckdb/common/types/data_chunk.hpp"
+#include "duckdb/common/pair.hpp"
+
+namespace duckdb {
+
+static void MapKeysFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	D_ASSERT(result.GetType().id() == LogicalTypeId::LIST);
+	idx_t count = args.size();
+
+	// Get the arguments vector
+	auto &map = args.data[0];
+	auto &map_keys = MapVector::GetKeys(map);
+
+	UnifiedVectorFormat map_keys_data;
+	UnifiedVectorFormat map_data;
+	map_keys.ToUnifiedFormat(count, map_keys_data);
+	map.ToUnifiedFormat(count, map_data);
+
+	auto &keys = ListVector::GetEntry(result);
+
+	D_ASSERT(keys.GetType().id() == ListType::GetChildType(map_keys.GetType()).id());
+
+	keys.Reference(ListVector::GetEntry(map_keys));
+
+	// Reference the data for the list_entry_t's
+	D_ASSERT(result.GetVectorType() == VectorType::FLAT_VECTOR);
+	FlatVector::SetData(result, map_keys_data.data);
+	FlatVector::SetValidity(result, map_data.validity);
+
+	auto list_size = ListVector::GetListSize(map_keys);
+	ListVector::SetListSize(result, list_size);
+
+	if (args.AllConstant()) {
+		result.SetVectorType(VectorType::CONSTANT_VECTOR);
+	}
+
+	result.Verify(count);
+}
+
+static unique_ptr<FunctionData> MapKeysBind(ClientContext &context, ScalarFunction &bound_function,
+                                            vector<unique_ptr<Expression>> &arguments) {
+	child_list_t<LogicalType> child_types;
+
+	if (arguments.size() != 1) {
+		throw InvalidInputException("The input argument must be a map");
```

Comment:
Shouldn't the error message say something like `too many parameters to function`. Imo there is a mismatch between the condition and the exception here, no?